lib/jobs/install-os.js

```diff
@@ -215,14 +217,40 @@ function installOsJobFactory(
 
     /**
      * @memberOf InstallOsJob
+     * Progress update after installation steps
+     */
+    InstallOsJob.prototype.updateProgress = function(descript, step) {
+        var progressData = {
+                graphId: this.context.graphId,
+                progress: {
+                    percentage: null,
+                    description: descript
+                },
+                taskProgress: {
+                    taskId: this.taskId,
+                    progress: {
+                        progressRate: step + '/' + this.options.steps,
+                        description: descript
+                    }
+                }
+        };
+        return TaskGraph.updateGraphProgress(progressData);
+    };
+
+    /**
+     * @memberOf InstallOsJob
      */
     InstallOsJob.prototype._run = function() {
         var self = this;
         return Promise.resolve().then(function() {
             return self._convertInstallDisk();
         }).then(function() {
             self._subscribeRequestProfile(function() {
-                return self.profile;
+                var description = "Reboot suceeded, starting initrd download.";
```

Comment:
At least for ESXi, no initrd. So please change to other common and reasonable words. Since the `kernel` is shared by all OS, you can use something like "starting boot kernel"